internal/fuzzbinder/fuzzclaim.go

```diff
@@ -0,0 +1,111 @@
+	// `ClaimSpec` per fuzz-target.
+	PerTarget []PerTargetSpec `json:"perTarget"`
+	// `ClaimSpec` for all the fuzz-targets.
+	PerProject *PerProjectSpec `json:"perProject"`
+}
+
+// PerTargetSpec contains the fuzzing claims specification per fuzz-target.
+type PerTargetSpec struct {
+	// Name of the fuzz-target.
+	Name string `json:"name"`
+	// URI of the fuzz-target.
+	URI string `json:"uri"`
+	// Coverage specifies the code coverage by the fuzz-target.
+	Coverage *FuzzCoverage `json:"coverage"`
+	// Bugs specifies the number of detected bugs using the fuzz-target.
+	Bugs int `json:"bugs"`
+	// Crashes specifies the number of crashes obtained using the fuzz-target.
+	Crashes int `json:"crashes"`
+	// FuzzEffort specifies the fuzzing efforts spent while using the fuzz-target.
+	FuzzEffort *FuzzEffortSpec `json:"fuzzEffort,omitempty"`
+}
+
+// PerProjectSpec contains the fuzzing claims specification of the revision
+// of the source code for all fuzz-targets.
+type PerProjectSpec struct {
+	// Coverage specifies the code coverage by all fuzz-targets.
+	Coverage *FuzzCoverage `json:"coverage"`
+	// Bugs specifies the  number of detected bugs using all fuzz-targets.
+	Bugs int `json:"bugs"`
+	// Crashes specifies the  number of crashes obtained using all fuzz-targets.
+	Crashes int `json:"crashes"`
+	// FuzzEffort specifies the fuzzing efforts spent while using all fuzz-targets.
+	FuzzEffort *FuzzEffortSpec `json:"fuzzEffort,omitempty"`
+}
+
+// FuzzCoverage contains the code coverage by fuzz testing.
+type FuzzCoverage struct {
+	// Line specifies line coverage.
+	Line string `json:"line"`
+	// Branch specifies branch coverage.
+	Branch string `json:"branch"`
+}
+
+// FuzzEffortSpec contains the fuzzing efforts.
+type FuzzEffortSpec struct {
+	// FuzzTime specifies the fuzzing time.
+	FuzzTime *time.Duration `json:"fuzzTime,omitempty"`
+	// NumberTests specifies the number of executed fuzzing tests.
+	NumberTests int `json:"numberTests,omitempty"`
+}
+
+// ValidateFuzzClaim validates that an Amber Claim is a Fuzz Claim with a valid ClaimType.
+// If valid, the ClaimPredicate object is returned. Otherwise an error is returned.
+func ValidateFuzzClaim(claimPredicate amber.ClaimPredicate) (*amber.ClaimPredicate, error) {
+	if claimPredicate.ClaimType != FuzzClaimV1 {
+		return nil, fmt.Errorf(
+			"the claimPredicate does not have the expected claim type; got: %s, want: %s",
+			claimPredicate.ClaimType,
+			FuzzClaimV1)
+	}
+
+	// Verify the type of the ClaimSpec, and return it if it is of type ClaimPredicate.
+	switch claimPredicate.ClaimSpec.(type) {
+	case FuzzClaimSpec:
+		return validateFuzzClaimSpec(claimPredicate)
+	default:
+		return nil, fmt.Errorf(
+			"the claimSpec does not have the expected type; got: %T, want: FuzzClaimSpec",
+			claimPredicate.ClaimSpec)
+	}
+}
+
+// validateFuzzClaimSpec validates details about the FuzzClaimSpec.
+func validateFuzzClaimSpec(claimPredicate amber.ClaimPredicate) (*amber.ClaimPredicate, error) {
+	// TBA
```

Comment:
You could open an issue and add //TODO(#issue number): Short description.

Reply:
I added `fmt.Println("Not yet implemented!")` to be clear that I will add it like for main.

Reply:
I added  `log.Println("Not yet implemented!")` to show that it is not yet implemented.